### DISCUSSION TITLE
x86 instructions to C?

### DISCUSSION THREAD

**Discussion by guest271314** [compiler-explorer/compiler-explorer#3441] [Q&A] [answered]
I have nexe file (x86 instructions). I want to decompile the file to C/C++. Is that possible?

**Comment by mattgodbolt** [comment, marked as answer]
It's possible but very tricky and error prone in general. Some commercial packages like HexRays can do it, albeit not to anything very human-readable without a lot of user help.  See https://stackoverflow.com/questions/14308610/how-hard-is-it-really-to-decompile-assembly-code for example.

It's not something Compiler Explorer does (nor any time likely) - we have a bug open for Ghidra support but I don't think Ghidra supports automated decompilation in the same way as HexRays.

**Reply by guest271314** [reply]
`hexdump` and `strings` give a peek https://gist.github.com/guest271314/4894662d31ffb4ad2cd57683721bf4b4.

**Reply by mattgodbolt** [reply]
Neither are anywhere near close to C though. If anything, the disassembly we show in CE is much nearer to "C" than either a hexdump, or a string list. If you want to disassemble an existing file, try `objdump -d`

**Reply by guest271314** [reply]
I tried that, too.

If I am not mistaken, even if I am able to disassemble and re-assemble the code, won't I still need the libraries referenced therein to re-compile the code?

I filed this request to Google https://github.com/GoogleChrome/developer.chrome.com/issues/2295.

**Reply by guest271314** [reply]
The source could very well be written in Go. I don't know. The source code is evidently restricted to "Googlers"

https://chromium.googlesource.com/chromium/src/+/refs/tags/74.0.3729.5/docs/accessibility/patts.md

> **Building from source**
> 
> This is for Googlers only.
> 
> Visit http://go/chrome-tts-blaze for instructions on how to build the engine from source and get the latest voice files.

**Comment by Amnezja81** [comment]
> I have nexe file (x86 instructions). I want to decompile the file to C/C++. Is that possible?

**Reply by mattgodbolt** [reply]
It's possible - but not on Compiler Explorer: see above.

There's now a site called dogbolt.org that provides a service more like the original poster wanted.
